libs/liquidity/src/lib/utils/liquidity-utils.ts

```diff
@@ -0,0 +1,122 @@
+import BigNumber from 'bignumber.js';
+import orderBy from 'lodash/orderBy';
+import { addDecimalsFormatNumber } from '@vegaprotocol/react-helpers';
+import { MarketState, MarketTradingMode } from '@vegaprotocol/types';
+import type {
+  LiquidityProvisionMarkets_marketsConnection_edges_node_tradableInstrument_instrument_product_settlementAsset as SettlementAsset,
+  LiquidityProvisionMarkets_marketsConnection_edges_node_candlesConnection_edges_node as Candle,
+  LiquidityProvisionMarkets_marketsConnection_edges_node as MarketNode,
+  LiquidityProvisionMarkets_marketsConnection_edges_node_liquidityProvisionsConnection_edges as LiquidityEdges,
+} from './../__generated__';
+import type { MarketCandles } from '@vegaprotocol/market-list';
+
+export type Market = MarketNode;
+
+export interface MarketsListData {
+  markets: Market[];
+  marketsCandles24hAgo: MarketCandles[];
+}
+
+const tradingModesOrdering = [
+  MarketTradingMode.TRADING_MODE_CONTINUOUS,
+  MarketTradingMode.TRADING_MODE_MONITORING_AUCTION,
+  MarketTradingMode.TRADING_MODE_BATCH_AUCTION,
+  MarketTradingMode.TRADING_MODE_OPENING_AUCTION,
+  MarketTradingMode.TRADING_MODE_NO_TRADING,
+];
+
+const filterMarkets = (markets: Market[]) =>
+  markets?.filter(
+    (m) =>
+      m.state !== MarketState.STATE_REJECTED &&
+      m.tradingMode !== MarketTradingMode.TRADING_MODE_NO_TRADING
+  );
+
+const orderMarkets = (markets: Market[]) =>
+  orderBy(markets, ['marketTimestamps.open', 'id'], ['asc', 'asc']);
+
+const sortMarkets = (markets: Market[]) =>
+  markets.sort(
+    (a, b) =>
+      tradingModesOrdering.indexOf(a.tradingMode) -
+      tradingModesOrdering.indexOf(b.tradingMode)
+  );
+
+// libs/market-list/src/lib/utils/market-utils.ts
```

Comment:
@mattrussell36 some functions in this file (where I've put the path references) are pretty much duplicates of existing functions but I can't reuse them because the types are not _exactly_ the same. That is, some libs pull in more fields than other and the functions uses the generated `types`, which means that I get TS errors when I try to reuse them. 
For example, row 38 here only requires an array of objects with a `tradingMode` key but `Market` is a generated type and has many more required fields. 
Is there a way to handle these things? To me, it would make sense to have a lib with some shared functions and simplified types, let me know if you think it's a good idea or an overkill.

Reply:
So I here you are spot on, where possible we should refactor these functions to declare their own types making them easier to work with across apps.

For example `calcDayVolume` takes an entire Market object when really we only need an array of partial candle objects.

```
const calcDayVolume = (market: Market) => BigNumber

// changes to

const caclDayVolume = (candles: Array<{ volume: string }>) => BigNumber
```

Reply:
PS this is something that is a bit messy in the repo as a whole. There is too much sharing of types between libs, ideally each lib will define its own types and its up to the app to fulfill the function's signature.

Reply:
@mattrussell36 Since I'm now reusing some providers some of these functions are no longer needed. The other ones I have refactored to declare their own types and only pass in the values that are actually needed 👍